physlr/physlr.py

```diff
@@ -430,6 +430,56 @@ def determine_backbones(g):
         print(int(timeit.default_timer() - t0), "Determined the backbone paths", file=sys.stderr)
         return backbones
 
+    @staticmethod
+    def wrap_up_messages_and_pass(mst, messages, sender, receiver):
+        """
+        Wrap up all incoming messages to this node (sender) except the one from receiver;
+        and set (pass) the message from sender to receiver.
+        """
+        neighbors_to_wrap_up = list(set(mst.neighbors(sender)) - {receiver})
```

Comment:
```suggestion
        neighbors_to_wrap_up = [u for u in mst.neighbors(sender) if u != receiver]
```

Reply:
This code is removing a single element from a list. Converting the list to a set (hashtable) and back to a list seems to me like an expensive way to accomplish that. You can use the `timeit` module to test intuition empirically.
```
>>> import timeit
>>> timeit.timeit("[x for x in range(100) if x != 50]", number=1_000_000)
5.630163773999811
>>> timeit.timeit("list(set(range(100)) - {50})", number=1_000_000)
4.746275222001714
>>> timeit.timeit("list(range(100)).remove(50)", number=1_000_000)
2.0266294670000207
```
So it turns out that list comprehension is *slower* than converting to a set and back (much to my surprise), and the fastest is using `list.remove`.